dipy/align/streamlinear.py

```diff
@@ -604,6 +643,330 @@ def bundle_min_distance_fast(t, static, moving, block_size, num_threads):
+        print('Moving streamlines size {}'.format(len(moving)))
+
+    def check_range(streamline, gt=greater_than, lt=less_than):
+
+        if (length(streamline) > gt) & (length(streamline) < lt):
+            return True
+        else:
+            return False
+
+    # TODO change this to the new Streamlines API
+    streamlines1 = [s for s in static if check_range(s)]
+    streamlines2 = [s for s in moving if check_range(s)]
+
+    if verbose:
+
+        print('Static streamlines after length reduction {}'
+              .format(len(streamlines1)))
+        print('Moving streamlines after length reduction {}'
+              .format(len(streamlines2)))
+
+    if select_random is not None:
+        rstreamlines1 = select_random_set_of_streamlines(streamlines1,
+                                                         select_random)
+    else:
+        rstreamlines1 = streamlines1
+
+    rstreamlines1 = set_number_of_points(rstreamlines1, nb_pts)
+    qb1 = QuickBundles(threshold=qb_thr)
+    rstreamlines1 = [s.astype('f4') for s in rstreamlines1]
+    cluster_map1 = qb1.cluster(rstreamlines1)
+    clusters1 = remove_clusters_by_size(cluster_map1, rm_small_clusters)
+    qb_centroids1 = [cluster.centroid for cluster in clusters1]
+
+    if select_random is not None:
+        rstreamlines2 = select_random_set_of_streamlines(streamlines2,
+                                                         select_random)
+    else:
+        rstreamlines2 = streamlines2
+
+    rstreamlines2 = set_number_of_points(rstreamlines2, nb_pts)
+    qb2 = QuickBundles(threshold=qb_thr)
+    rstreamlines2 = [s.astype('f4') for s in rstreamlines2]
+    cluster_map2 = qb2.cluster(rstreamlines2)
+    clusters2 = remove_clusters_by_size(cluster_map2, rm_small_clusters)
+    qb_centroids2 = [cluster.centroid for cluster in clusters2]
+
+    if verbose:
+        t = time()
+
+    if not progressive:
+        slr = StreamlineLinearRegistration(x0=x0,
+                                           options={'maxiter': maxiter},
+                                           num_threads=num_threads)
+        slm = slr.optimize(qb_centroids1, qb_centroids2)
+    else:
+        bounds = DEFAULT_BOUNDS
+
+        slm = progressive_slr(qb_centroids1, qb_centroids2,
+                              x0=x0, metric=None,
+                              bounds=bounds, num_threads=num_threads)
+
+    if verbose:
+        print('QB static centroids size %d' % len(qb_centroids1,))
+        print('QB moving centroids size %d' % len(qb_centroids2,))
+        duration = time() - t
+        print('SLR finished in  %0.3f seconds.' % (duration,))
+        if slm.iterations is not None:
+            print('SLR iterations: %d ' % (slm.iterations,))
+
+    moved = slm.transform(moving)
+
+    return moved, slm.matrix, qb_centroids1, qb_centroids2
+
+
+# In essence whole_brain_slr can be thought as a combination of
+# SLF and QuickBundles
+whole_brain_slr = slr_with_qb
```

Comment:
whole_brain_slr and slr_with_qb basically have same implementation?

Reply:
Yes, the same function can be used for brain tractograms and other tractograms. In the paper I called it `whole_brain_slr` but then realized that the function can be more generic. Adding some more documentation to explain.